Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.cpp

```diff
@@ -148,14 +153,25 @@ OptionSet<CoordinatedBackingStoreProxy::UpdateResult> CoordinatedBackingStorePro
         result.add(UpdateResult::TilesPending);
 
     // Update the dirty tiles.
+    IntRect tileDirtyRectUnion;
     unsigned dirtyTilesCount = 0;
     for (const auto& tile : m_tiles.values()) {
-        if (tile.isDirty())
-            dirtyTilesCount++;
+        if (!tile.isDirty())
+            continue;
+
+        tileDirtyRectUnion.unite(tile.dirtyRect);
+        ++dirtyTilesCount;
     }
 
     WTFBeginSignpost(this, UpdateTiles, "dirty tiles: %u", dirtyTilesCount);
 
+#if USE(SKIA)
+    // Record only once the whole layer.
+    std::optional<Ref<SkiaRecordingResult>> recording;
+    if (LIKELY(dirtyTilesCount > 0 && layer.client().paintingEngine().useThreadedRendering()))
```

Comment:
I don't like that the backing store proxy needs to know about threaded rendering, but don't know a better solution. The other option would be that the paint engine knows about tiles, and that's not better.

Reply:
I came to the same conclusion, it's not pretty either way. At least it's minimal invasive this way.